python/lsst/pipe/base/pipeline.py

```diff
@@ -451,11 +453,6 @@ def fromPipeline(cls, pipeline: pipelineIR.PipelineIR) -> Pipeline:
         return cls.fromIR(copy.deepcopy(pipeline._pipelineIR))
 
     def __str__(self) -> str:
-        # tasks need sorted each call because someone might have added or
-        # removed task, and caching changes does not seem worth the small
-        # overhead
-        labels = [td.label for td in self._toExpandedPipelineImpl(checkContracts=False)]
-        self._pipelineIR.reorder_tasks(labels)
         return str(self._pipelineIR)
```

Comment:
Back when we were discussing your sorted-pipeline ticket, you said you'd originally not changed the order because you thought the preserving the input order was more important, and I had to change your mind (and that argument centered on being able to compare pipelines).  I figured that now that we had a different way to write pipelines so that they could be compared, leaving this as a way to get the input order back unchanged would be better - especially because importing all of the tasks and instantiating all of the connection classes just to sort can actually be pretty slow, and then we throw that all away because we can't get away with caching the `TaskDefs` inside the `Pipeline`.

Reply:
I think you so successfully argued for comparing things before you completely changed my mind. That said your argument now is pretty good and I think one of the things I was thinking before. So basically I think I am fine with anything :)